python-lib/dku_deeplearning_image/utils.py

```diff
@@ -164,33 +171,75 @@ def wrapper(*args, **kwargs):
-    if not classify:
-        return predictions.tolist()
+def format_predictions_output(predictions, errors, classify=False, labels_df=None, limit=None, min_threshold=None):
     formatted_predictions = []
+    predictions = list(predictions)
     id_pred = lambda index: labels_df.loc[index][constants.LABEL] if labels_df is not None else str(index)
-    for pred in predictions:
-        formatted_pred = get_ordered_dict(
-            {id_pred(i): float(pred[i]) for i in pred.argsort()[-limit:] if float(pred[i]) >= min_threshold})
-        formatted_predictions.append(formatted_pred)
-    return formatted_predictions
+    for err_i in range(len(errors)):
+        if errors[err_i] == 1:
+            predictions.insert(err_i, None)
+    for pred_i, pred in enumerate(predictions):
+        if pred is not None:
+            if classify:
+                formatted_pred = OrderedDict(
+                    [(id_pred(i), float(pred[i])) for i in pred.argsort()[-limit:] if float(pred[i]) >= min_threshold])
+                formatted_predictions.append(json.dumps(formatted_pred))
+            else:
+                formatted_predictions.append(pred.tolist())
+        else:
+            logger.warning(f"There has been an error with prediction: {pred_i}. (It is probably not an image)")
+            formatted_predictions.append(None)
+    return {"prediction": formatted_predictions, "error": errors}
+
 
+def apply_preprocess_image(tfds, input_shape, preprocessing, is_b64=False):
+    def _apply_preprocess_image(image_path):
+        return tf.numpy_function(
+            func=lambda x: preprocess_img(x, input_shape, preprocessing, is_b64),
+            inp=[image_path],
+            Tout=tf.float32)
 
-def get_predictions(model, batch, classify=False, limit=constants.DEFAULT_PRED_LIMIT, min_threshold=0, labels_df=None):
-    predictions = model.predict(batch)
-    return format_predictions_output(predictions, classify, labels_df, limit, min_threshold)
+    def _convert_errors(images):
+        return tf.numpy_function(
+            func=lambda x: tf.cast(x.size == 0, tf.int8),
+            inp=[images],
+            Tout=tf.int8)
 
+    def _filter_errors(images):
+        return tf.numpy_function(
+            func=lambda x: x.size != 0,
+            inp=[images],
+            Tout=tf.bool)
 
-def get_ordered_dict(predictions):
-    return json.dumps(OrderedDict(sorted(predictions.items(), key=(lambda x: -x[1]))))
+    preprocessed_images = tfds.map(map_func=_apply_preprocess_image, num_parallel_calls=constants.AUTOTUNE)
+    error_array = preprocessed_images.map(map_func=_convert_errors, num_parallel_calls=constants.AUTOTUNE)
+    preprocessed_images_filtered = preprocessed_images.filter(predicate=_filter_errors)
 
+    return preprocessed_images_filtered, error_array
 
-def preprocess_img(img_path, img_shape, preprocessing):
+
+def read_images_to_tfds(images_folder, np_images):
```

Comment:
`read_images` does not actually read the images, but fetches them from the managed folder and yields the path to the image, so the name is confusing

Reply:
Fixed in 4f9a704